sos/report/plugins/aws.py

```diff
@@ -41,21 +41,19 @@ def setup(self):
 
         # Try to get an IMDSv2 token
         token_url = 'http://169.254.169.254/latest/api/token'
-        token_cmd = [
-            'curl', '-sS', '-X', 'PUT', '-H',
-            'X-aws-ec2-metadata-token-ttl-seconds: 21600',
-            token_url]
+        token_cmd = 'curl -sS -X PUT -H \
+            X-aws-ec2-metadata-token-ttl-seconds:21600 {}'.format(token_url)
 
         try:
-            token = self.exec_cmd(token_cmd, timeout=1)
+            token = self.exec_cmd(token_cmd, timeout=1)['output']
         except Exception:
             token = ''
 
         # Add header only if token retrieval succeeded
-        token_header = []
+        token_header = ''
 
         if token:
-            token_header = ['-H', f'X-aws-ec2-metadata-token: {token}']
+            token_header = '-H X-aws-ec2-metadata-token:{}'.format(token)
```

Comment:
same here - f-strings if possible, unless you have a good reason to prefer .format()

Reply:
Thanks, updated to f-strings.